worlds/zillion/__init__.py

```diff
@@ -0,0 +1,392 @@
+        zz_patcher.all_fixes_and_options(zz_options)
+        zz_patcher.set_external_item_interface(zz_options.start_char, zz_options.max_level)
+        zz_patcher.set_multiworld_items(multi_items)
+        zz_patcher.set_rom_to_ram_data(self.world.player_name[self.player].replace(' ', '_').encode())
+
+    def generate_output(self, output_directory: str) -> None:
+        """This method gets called from a threadpool, do not use world.random here.
+        If you need any last-second randomization, use MultiWorld.slot_seeds[slot] instead."""
+        self.finalize_item_locations()
+
+        assert self.zz_system.patcher, "didn't get patcher from generate_early"
+        # original_rom_bytes = self.zz_patcher.rom
+        patched_rom_bytes = self.zz_system.patcher.get_patched_bytes()
+
+        out_file_base = self.world.get_out_file_name_base(self.player)
+
+        filename = os.path.join(
+            output_directory,
+            f'{out_file_base}{ZillionDeltaPatch.result_file_ending}'
+        )
+        with open(filename, "wb") as binary_file:
+            binary_file.write(patched_rom_bytes)
+        patch = ZillionDeltaPatch(
+            os.path.splitext(filename)[0] + ZillionDeltaPatch.patch_file_ending,
+            player=self.player,
+            player_name=self.world.player_name[self.player],
+            patched_path=filename
+        )
+        patch.write()
+        os.remove(filename)
+
+    def fill_slot_data(self) -> Dict[str, Any]:  # json of WebHostLib.models.Slot
+        """Fill in the `slot_data` field in the `Connected` network package.
+        This is a way the generator can give custom data to the client.
+        The client will receive this as JSON in the `Connected` response."""
+
+        # TODO: share a TypedDict data structure with client
+
+        # TODO: tell client which canisters are keywords
+        # so it can open and get those when restoring doors
+
+        zz_patcher = self.zz_system.patcher
+        assert zz_patcher, "didn't get patcher from generate_early"
+        assert self.zz_system.randomizer, "didn't get randomizer from generate_early"
+
+        rescues: Dict[str, Any] = {}
+        for i in (0, 1):
+            if i in zz_patcher.rescue_locations:
+                ri = zz_patcher.rescue_locations[i]
+                rescues[str(i)] = {
+                    "start_char": ri.start_char,
+                    "room_code": ri.room_code,
+                    "mask": ri.mask
+                }
+        zz_patcher.loc_memory_to_loc_id
+        return {
+            "start_char": self.zz_system.randomizer.options.start_char,
+            "rescues": rescues,
+            "loc_mem_to_id": zz_patcher.loc_memory_to_loc_id
+        }
+
+    def modify_multidata(self, multidata: Dict[str, Any]) -> None:
+        """For deeper modification of server multidata."""
+        # not modifying multidata, just want to call this at the end of the generation process
+        clear_cache()
```

Comment:
this breaks the simulated playthrough

Reply:
Does it? Wouldn't it just repopulate the cache?

Is there someplace I can put this to run it after all logic stuff is done?

Reply:
attached logic cache to multiworld